scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/FileSystemService.java

```diff
@@ -45,52 +44,51 @@ public void addDatasetDirectories(String id, String username) throws DatasetStor
                 Files.createDirectories(path);
             }
         } catch (IOException e) {
-            throw new DatasetStorageException("IOException occurred when attempting to add the Dataset", e);
+            throw new FileSystemStorageException("IOException occurred when attempting to add the Dataset", e);
         } catch (UnsupportedOperationException e) {
-            throw new DatasetStorageException("UnsupportedOperationException occurred when attempting to add the Dataset", e);
+            throw new FileSystemStorageException("UnsupportedOperationException occurred when attempting to add the Dataset", e);
         } catch (SecurityException e) {
-            throw new DatasetStorageException("SecurityException occurred when attempting to add the Dataset", e);
+            throw new FileSystemStorageException("SecurityException occurred when attempting to add the Dataset", e);
         }
     }
 
     /**
      * A helper method to create a Path given a username and dataset name.
      *
      * @param username  The username of the owner of the dataset.
-     * @param datasetId The id of the dataset.
+     * @param directory The directory name.
      * @return Returns a Path.
+     * @throws InvalidPathException Exception if the path cannot be created.
      */
     @Override
-    public Path createPath(String username, String datasetId) {
+    public Path createPath(String username, String directory) throws InvalidPathException {
         Path path;
-        try {
-            if (datasetId != null) {
-                path = Paths.get(this.rootLocationString, username, datasetId);
-            } else {
-                path = Paths.get(this.rootLocationString, username);
-            }
-            return Paths.get(StringUtils.cleanPath(Objects.requireNonNull(path).toString()));
-        } catch (InvalidPathException e) {
-            throw new DatasetStorageException(
-                    "An InvalidPathException occurred in createPath with null datasetName", e);
+        if (directory != null) {
+            path = Paths.get(this.rootLocationString, username, directory);
+        } else {
+            path = Paths.get(this.rootLocationString, username);
         }
+        return Paths.get(StringUtils.cleanPath(Objects.requireNonNull(path).toString()));
     }
 
     /**
      * Deletes a dataset on the filesystem.
      *
      * @param id The id of the dataset.
-     * @throws SecurityException throws a storage exception if the directory could not be deleted.
+     * @throws FileSystemStorageException throws an exception if the directory could not be deleted.
      */
     @Override
-    public void deleteDatasetDirectories(Long id, String username) throws SecurityException {
+    public void deleteDatasetDirectories(Long id, String username) throws FileSystemStorageException {
```

Comment:
Similar to above, this doesn't necessarily have to be a dataset directory, right? This method would delete a directory in a user's folder no matter what it is used for. I would not pass a long but a string (so you can use it more easily later for other things) and rename the method.